### PR TITLE
[mcore] fix: support FP8 padding for BSHD actor forward

### PR DESCRIPTION
**Pull request by gem-mint**
### What does this PR do?

Enables Megatron actor/ref forward passes in BSHD format to run with Transformer Engine FP8 enabled.

Qwen3.x/GDN-style models can require `use_remove_padding=False`, which routes Megatron actor updates through the BSHD forward path. Before this change, that path rejected FP8 with an assertion and did not apply the FP8 block-alignment padding that TE kernels require.

### Root cause

`model_forward_gen(..., data_format="bshd")` explicitly asserted that `fp8 is None`, so actor update/log-prob computation failed as soon as `override_transformer_config.fp8=hybrid` or `e4m3` was enabled.

After removing the assertion, BSHD still needed the same kind of FP8-aware sequence padding used by the packed/no-padding paths: TE block FP8 kernels require the local token extent to be divisible by 128 after TP/CP partitioning.

### Design & Code Changes

- Remove the BSHD FP8 assertion in `verl/models/mcore/model_forward.py`.
- Pass `use_fp8_padding` through BSHD input preprocessing and logits-processor argument preprocessing.
- Add `_align_bshd_max_seqlen_for_fp8` in `verl/models/mcore/util.py` to align BSHD sequence length for TE FP8 block quantization while preserving TP/CP divisibility constraints.
- Reuse the helper in `preprocess_bshd_no_padding` to keep BSHD FP8 alignment logic consistent.
- Add a CPU regression test covering BSHD FP8 padding shape/alignment and postprocess round-trip behavior.

### Duplicate-work check

No linked issue was provided for this fix.

Checks run before opening this PR:

```bash
gh issue list --repo verl-project/verl --state open --search '"fp8 is not supported for bshd format yet"' --limit 20
gh issue view 5508 --repo verl-project/verl --comments
gh pr list --repo verl-project/verl --state open --search '"fp8 is not supported for bshd format yet"' --limit 20
gh pr list --repo verl-project/verl --state open --search 'bshd fp8' --limit 20
gh pr list --repo verl-project/verl --state open --search 'Megatron BSHD FP8' --limit 20
gh pr list --repo verl-project/verl --state open --search 'actor fp8 padding' --limit 20
gh pr list --repo verl-project/verl --state open --search 'bshd actor fp8' --limit 20
```

Results:

- Issue #5508 is about vision-model BSHD usage and recommends THD/remove-padding for VLMs; it does not address Megatron actor BSHD FP8 support.
- PR #6703 fixes a separate `preprocess_thd_engine` FP8 + CP out-of-bounds issue for THD/SFT. This PR targets BSHD actor forward/logprob padding and removes the BSHD FP8 assertion, so it is not a duplicate.
- The actor-specific FP8 padding searches returned no open PRs.

### Test

Ran successfully:

```bash
git diff --check
python -m py_compile verl/models/mcore/util.py verl/models/mcore/model_forward.py tests/models/test_mcore_bshd_fp8_padding.py
```

Attempted but not run in this local Windows runtime:

```bash
python -m pytest tests/models/test_mcore_bshd_fp8_padding.py -q
# failed locally because pytest is not installed in this runtime

pre-commit --version
# failed locally because pre-commit is not installed in this runtime
```

External validation: the patched v0.7.1 tree was used to start a Qwen3.x-35B-A3B GRPO training run with Megatron + SGLang and actor FP8 enabled.

### AI assistance

This PR was prepared with AI assistance. Human review done before marking the PR ready for review.

### CODE REVIEW

**Comment by gem-mint** [comment]
Closing in favor of #6884, which uses the same commit but removes `codex` from the head branch name.
